modules/rules_nodejs/4.4.6/patches/extensions.patch

```diff
@@ -0,0 +1,61 @@
+commit a3e63c15436edd1eac2646470034ac7f8a91dc2c
+Author: Alex Eagle <alex@aspect.dev>
+Date:   Sat Dec 11 15:12:06 2021 -0800
+
+    feat: allow usage from bzlmod
+    
+    Fixes #3138
+
+diff --git a/nodejs/extensions.bzl b/nodejs/extensions.bzl
+new file mode 100644
+index 00000000..9b57598e
+--- /dev/null
++++ b/nodejs/extensions.bzl
+@@ -0,0 +1,24 @@
++"extensions for bzlmod"
++
++load(":repositories.bzl", "nodejs_register_toolchains")
++
++node_toolchain = tag_class(attrs = {
++    "name": attr.string(doc = "Base name for generated repositories"),
++    "node_version": attr.string(doc = "Version of the Node.js interpreter"),
++})
++
++def _toolchain_extension(module_ctx):
++    # according to docs, modules iteration order is guaranteed to
++    # be the same as breadth-first search starting from the root module.
++    for mod in module_ctx.modules:
++        for toolchain in mod.tags.toolchain:
```

Comment:
I'd suggest you to collect all the toolchains into a dict before creating repos for them, so that two modules calling "node.toolchain" with the same name wouldn't cause you to try to create two repos with the same name.

Reply:
wouldn't it be better to error if the user tries to declare conflicting names for toolchains?

Reply:
yes it would; I was concerned with the case where two unrelated modules declare toolchains with the same name. Say you're module "foo" and you declare toolchain A, and you depend on module "bar" and it also declares a toolchain A with the same node version, you would get an error because the extension tries to create two repos called "node_toolchain_A", when it's actually okay.

You could just decide that this case is not worth caring about -- up to you :)

Reply:
Updated the logic - I allow duplicate registrations with the same node version, but error if they conflict.

Reply:
nice!